src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Window/PlatformWindowEffect.android.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.ComponentModel;
+using Android.OS;
+using Android.Views;
+using Xamarin.CommunityToolkit.Android.Effects;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using Color = Android.Graphics.Color;
+
+[assembly: ExportEffect(typeof(PlatformWindowEffect), nameof(WindowEffect))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	public class PlatformWindowEffect : PlatformEffect
+	{
+		protected override void OnAttached()
+		{
+			if (WindowEffect.GetStatusBarColor(Element) != (Forms.Color)WindowEffect.StatusBarColorProperty.DefaultValue)
+			{
+				SetStatusBarColor(WindowEffect.GetStatusBarColor(Element).ToAndroid());
+			}
+			if (WindowEffect.GetStatusBarStyle(Element) != (StatusBarStyle)WindowEffect.StatusBarStyleProperty.DefaultValue)
+			{
+				SetStatusBarStyle(WindowEffect.GetStatusBarStyle(Element));
+			}
+		}
+
+		protected override void OnDetached()
+		{
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+			if (args.PropertyName == WindowEffect.StatusBarColorProperty.PropertyName)
+			{
+				SetStatusBarColor(WindowEffect.GetStatusBarColor(Element).ToAndroid());
+			}
+			else if (args.PropertyName == WindowEffect.StatusBarStyleProperty.PropertyName)
+			{
+				SetStatusBarStyle(WindowEffect.GetStatusBarStyle(Element));
+			}
+		}
+
+		public void SetStatusBarColor(Color color)
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+				return;
+
+			Activity.SetStatusBarColor(color);
+		}
+
+		public void SetStatusBarStyle(StatusBarStyle style)
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+				return;
+
+			switch (style)
+			{
+				case StatusBarStyle.Default:
+				case StatusBarStyle.LightContent:
+					RemoveBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
```

Comment:
is StatusBarVisibility deprecated for >= API 30?
https://medium.com/swlh/modifying-system-ui-visibility-in-android-11-e66a4128898b
https://stackoverflow.com/questions/63947770/how-can-i-make-the-status-bar-white-with-black-icons-android-11

Reply:
It is, but since XCT is targeting Android 10, we cannot use Android 11s SDK yet, also because we are targeting Android 10, this logic works on Android 11 devices (tested this).

We had a discussion about this with @pictos [here](https://github.com/xamarin/XamarinCommunityToolkit/pull/812#discussion_r575855668).

Reply:
Added note about this in PR description